contracts/FIFOMarket.sol

```diff
@@ -182,10 +190,15 @@ contract FIFOMarket is
       "Market: This contract can only receive BridgedPolygonNORI"
     );
 
+    uint256 certificateAmount = (amount * 100) / (100 + _noriFee); // todo calculate noriFee in loop to allow dynamic fees per removal
+    uint256 remainingAmountToFill = certificateAmount;
     uint256[] memory ids = new uint256[](_queueLength());
     uint256[] memory amounts = new uint256[](_queueLength());
     address[] memory suppliers = new address[](_queueLength());
     uint256 numberOfRemovals = 0;
+    Fee memory noriFee = Fee(_noriFeeWallet, 0, 0, true); // todo throw if supplier address == nori fee address
```

Comment:
I actually went down a rabbit hole on this one yesterday and tried implementing per token royalties using [EIP 2891](https://docs.openzeppelin.com/contracts/4.x/api/token/common#ERC2981). I ultimately ended up dropping that from this PR as it would have introduced additional complexity and I wasn't even sure if we actually did want to introduce per-removal fee variables. Per-removal variable fees was actually pretty hard to figure out as well since we actually need to send the an amount of tokens + the fee when calling `bpNori.send`. The only way to do that would be to pre-calculate the fee and pass it in as data but that would also ultimately be prone to errors/bugs due to the race conditions it would introduce off-chain

Reply:
oh right, I remember thinking about this problem and ran into the same issue about having to know the total overall fee percentage for the order to be able to calculate how many NRTs are actually getting purchased v.s. what's going to fees.  Not only is it super challenging to implement from a race condition perspective (one idea I had would be that that part of the data you pass in to send are exactly how many NRTs you want to buy, and then you submit a payment that includes an upper bound on the fees, and then you get refunded whatever doesn't go to fees) - but I feel like we should never do this because it's effectively charging different buyers different prices for NRTs, which doesn't make any good business sense at all to me.